### PR TITLE
Storage init / Improvments

### PR DESCRIPTION
**Pull request by Tropaul322**
Done: 
1. Initialized storage for saving notes
2. Initialized background worker to send current url of page (Need improvements)
3. Fetch/Display notes from storage
4. Add ability to drag notes
5. Made improvements to Note component 
6. Comment/Delete unnessery code to build faster
7. Action to remove all notes from a page
8. Maybe stmh else but i forgot 🙃

### CODE REVIEW

File: src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -14,11 +14,25 @@ const Menu: React.FC<{}> = () => {
...
+      chrome.tabs.sendMessage(tabs[0].id, { type: Messages.DELETE_ALL });
+    }
+  };
   return (
     <StyledList>
       <StyledListItem onClick={handleClick}>
         <StickyNote2Icon fontSize="small" /> Create a new note
       </StyledListItem>
+      <StyledListItem onClick={deleteAllNotes}>
+        <StickyNote2Icon fontSize="small" /> Delete all notes from page
```
Comment (staceysych23) [inline src/components/Menu/Menu.tsx]: we might want another icon?

File: src/components/Note/Note.styled.ts [hunk 2]
```diff
@@ -19,20 +22,20 @@ export const StyledCard = styled(Card)<StyledCardProps>`
...
+  top: ${(props) => props.coords.top }px;
+  left: ${(props) => props.coords.left }px;
 `;
 
 export const StyledHeader = styled(Box)<StyledHeaderProps>`
   display: flex;
   align-items: center;
   justify-content: space-between;
-  background-color: ${(props) => props.background};
+  background-color: ${(props) => colorPalette[props.background].main};
```
Comment (staceysych23) [inline src/components/Note/Note.styled.ts:38]: I like this change :)

File: src/contentScript/contentScript.tsx [hunk 3]
```diff
@@ -2,17 +2,30 @@ import React, { useEffect, useState } from "react";
...
+    }
+    if (type === Messages.NEW_PAGE) {
+      setCurrentPageUrl(data.url);
+      const notes = await fetchNotes(data.url);
+
+      setNotes(notes);
+    }
+    if (type === Messages.DELETE_ALL) {
+      chrome.storage.sync.remove(currentPageUrl);
+      setNotes([]);
```
Comment (staceysych23) [inline src/contentScript/contentScript.tsx]: can we do it using switch?

File: src/options/options.css [hunk 4]
```diff
@@ -1,4 +1,4 @@
-body {
+/* body {
   background-color: #f5f5f5;
   font-family: 'Roboto';
-}
+} */
```
Comment (staceysych23) [inline src/options/options.css]: why do you comment this out? If we don't need this file, should we better delete it?

File: src/utils/storage.ts [hunk 5]
```diff
@@ -1,54 +1,40 @@
...
-  hasAutoOverlay: boolean
-  homeCity: string
-  tempScale: OpenWeatherTempScale
-}
+export const fetchNotes = async (url: string): Promise<any[]> => {
+  return new Promise((resolve) => {
+    chrome.storage.sync.get([url], (notes) => {
+      resolve(notes[url] ? JSON.parse(notes[url]) : []);
+    });
+  });
```
Comment (staceysych23) [inline src/utils/storage.ts:12]: why do we need to use Promise here?

File: src/components/Menu/Menu.tsx (hunk 1 above)
Comment (Tropaul322) [reply src/components/Menu/Menu.tsx]: It's just a helper for us, we might delete it later

File: src/components/Note/Note.styled.ts (hunk 2 above)
Comment (Tropaul322) [reply src/components/Note/Note.styled.ts:38]: Thx

File: src/contentScript/contentScript.tsx (hunk 3 above)
Comment (Tropaul322) [reply src/contentScript/contentScript.tsx]: yep will do, and i thinking to make separate functions for each case but i dont know if we can pass useState func as params

File: src/options/options.css (hunk 4 above)
Comment (Tropaul322) [reply src/options/options.css]: I don't know what it is, so i just commened everything so we could use it as reference later

File: src/utils/storage.ts (hunk 5 above)
Comment (Tropaul322) [reply src/utils/storage.ts:12]: Copied from tutorial, maybe we don't need it here, will check

Comment (Tropaul322) [reply src/utils/storage.ts:12]: UPD: Promise helps to return data from a callback

File: src/contentScript/contentScript.tsx (hunk 3 above)
Comment (staceysych23) [reply src/contentScript/contentScript.tsx]: we can pass setSomethig function if we need

**Pull request merged**